### PR TITLE
Don't pass InvalidTransactionId to update_next_xid.

### PR DESCRIPTION
**Pull request by hlinnaka**
update_next_xid() doesn't have any special treatment for the invalid or other special XIDs, so it will treat InvalidTransactionId (0) as a regular XID. If old nextXid is smaller than 2^31, 0 will look like a very old XID, and nothing happens. But if nextXid is greater than 2^31 0 will look like a very new XID, and update_next_xid() will incorrectly bump up nextXID.

### CODE REVIEW

**Review by kelvich — APPROVED** [review]

**Comment by kelvich** [comment]
```
 error: expected one of `.`, `;`, `?`, `}`, or an operator, found `{`
   --> pageserver/src/walingest.rs:106:61
    |
106 |             self.checkpoint.update_next_xid(decoded.xl_xid) {
    |                                                             ^ expected one of `.`, `;`, `?`, `}`, or an operator
```

File: pageserver/src/walingest.rs [hunk 1]
```diff
@@ -330,8 +332,13 @@ impl WalIngest {
...
                     {
                         self.checkpoint.oldestXid = xlog_checkpoint.oldestXid;
-                        self.checkpoint_modified = true;
                     }
+
+                    // Write a new checkpoint key-value pair on every checkpoint record, even
+                    // if nothing really changed. Not strictly required, but it seems nice to
+                    // have some trace of the checkpoint records in the layer files at the same
+                    // LSNs.
+                    self.checkpoint_modified = true;
```
Comment (hlinnaka) [inline pageserver/src/walingest.rs:341]: I'd like to have some eyeballs on this change. I think this is a good idea and should not cause any harm, but am I missing some unforeseen consequences?

This is kind of a similar issue as https://github.com/neondatabase/neon/issues/5962. With this, a checkpoint record is no longer a no-op as far as the pageserver is concerned.

Comment (hlinnaka) [reply pageserver/src/walingest.rs:341]: @arssher, @problame

Comment (arssher) [reply pageserver/src/walingest.rs:341]: Looks good, it may somewhat improve situation with #5962 , but can't harm anything AFAIS.

**Comment by hlinnaka** [comment]
One test failure in the e2e public API tests: https://github.com/neondatabase/cloud/actions/runs/7595014557/job/20687120459. I'm pretty sure that's a flaky test, because all the other variants of the e2e tests failed, and they don't exercise this codepath in any special manner anyway.

**Comment by hlinnaka** [comment]
The e2e test suite failed again, at a different test. I still believe it's flakiness, but would appreciate a second opinion from someone more in touch with the e2e tests

**Pull request merged**